### PR TITLE
Connector connection recovery improvement

### PR DESCRIPTION
**Pull request by abuijze**
This PR addresses a race condition that can occur when setting up a connection that fails in a very short period of time. In such a case, the connector could believe it still has a valid connection with Axon Server.

Logging has been improved to include the name of context for which the message applies.

### CODE REVIEW

File: src/main/java/io/axoniq/axonserver/connector/impl/AbstractIncomingInstructionStream.java
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/impl/AbstractIncomingInstructionStream.java:64]: I would've assumed `connection` instead of `call` in the second sentence here, although that might be a wrong assumption from my end.

File: src/main/java/io/axoniq/axonserver/connector/impl/ControlChannelImpl.java
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/impl/ControlChannelImpl.java:172]: Why use a method `registerOutboundStream` in the `CommandChannelImpl` and a lambda in this class?

File: src/main/java/io/axoniq/axonserver/connector/query/impl/QueryChannelImpl.java
Comment (smcvb) [inline src/main/java/io/axoniq/axonserver/connector/query/impl/QueryChannelImpl.java:202]: Why use a method `registerOutboundStream` in the `CommandChannelImpl` and a lambda in this class?

**Review by smcvb — COMMENTED** [review]
Some small pointers.

**Review by m1l4n54v1c — APPROVED** [review]
LGTM! Once @smcvb's requests are done, you can merge it.

**Review by smcvb — APPROVED** [review]
My concerns have been addressed, hence approving.

**Pull request merged**
